### ISSUE TITLE
TST 56 tests in pandas/tests/test_generic.py aren't being tested

### ISSUE THREAD

**Issue by MarcoGorelli** [pandas-dev/pandas#50380]
All these tests aren't actually being run:

https://github.com/pandas-dev/pandas/blob/2ffb74814fb7ba800ed4ec6bc1935ccb78f9cb65/pandas/tests/generic/test_generic.py#L53-L306

that's because the class name is `Generic`, rather than `TestGeneric`

They're all uncovered in the coverage report: https://app.codecov.io/gh/pandas-dev/pandas/blob/main/pandas/tests/generic/test_generic.py

Similar to https://github.com/pandas-dev/pandas/issues/50378, the class needs renaming, and the tests fixing up

**Comment by phershbe** [comment]
take

**Comment by phershbe** [comment]
@MarcoGorelli I tried `take` but I didn't realize that that was exclusively for `good first issue`. I'm interested in working on this. I've changed the class name and run the tests and got 20 failing, but many of the error messages are similar to each other. I'm digging in now to understand what they do, the error messages are helpful obviously.

**Comment by MarcoGorelli** [comment]
🤔 I thought `take` worked for any issue, not sure what happened here - consider it yours anyway

and thanks for working on this!

**Comment by phershbe** [comment]
@MarcoGorelli I'm working on it and learning a lot. I wasn't sure that I knew enough programming to get it, but after some digging, things are starting to make sense. I learned what the `parametrize` decorator does, where `frame_or_series` was coming from and then what `conftest.py` is what a `fixture` does, et cetera. I'm going to work on it a lot today too, if I start getting really stuck I'll let you know.

**Comment by phershbe** [comment]
@MarcoGorelli I've spent a long time on this and I've learned a ton and can read the code in the tests much better now, and understand much more about Pandas and even NumPy.

I was writing here to ask some questions and to say that I know that I can't spend forever on this and that I would unassign soon if I was still working through things, but then I got `test_rename` to pass by adding `frame_or_series` as the `box` parameter in `construct` on line 70. I'm learning a lot about how to understand unfamiliar code by breaking it down into many small pieces, which is much harder than the tutorials that I was doing and simple projects that I was making, obviously.

I don't know if the test is correct now, but I'm very encouraged that I at least got it to pass. If it's okay, let me spend the rest of the day to try to get the other tests to pass, and I'll get a pull request in tomorrow morning or leave a message and somebody else can do it if necessary. Thank you.
